crypto/crypto-derive/src/lib.rs

```diff
@@ -145,19 +145,26 @@ pub fn silent_debug(source: TokenStream) -> TokenStream {
 pub fn deserialize_key(source: TokenStream) -> TokenStream {
     let ast: DeriveInput = syn::parse(source).expect("Incorrect macro input");
     let name = &ast.ident;
+    let name_string = name.to_string();
     let gen = quote! {
         impl<'de> ::serde::Deserialize<'de> for #name {
             fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
             where
                 D: ::serde::Deserializer<'de>,
             {
+                #[derive(::serde::Deserialize)]
+                #[serde(rename = #name_string)]
+                struct _Value<'a>(&'a[u8]);
```

Comment:
should this be in the else statement does it matter? it might also be good to leave a comment so that some enterprising individual doesn't break this... or better yet add a test

Reply:
Good point. There will be a non-regression test soon. I also left a comment.